pkg/registry/controller/etcd/etcd_test.go

```diff
@@ -219,217 +211,68 @@ func TestEtcdListControllers(t *testing.T) {
 		})
 }
 
-func TestEtcdDeleteController(t *testing.T) {
-	ctx := api.NewDefaultContext()
+func TestWatchControllers(t *testing.T) {
 	storage, fakeClient := newStorage(t)
-	key, _ := storage.KeyFunc(ctx, validController.Name)
-	key = etcdtest.AddPrefix(key)
-
-	fakeClient.Set(key, runtime.EncodeOrDie(testapi.Codec(), validNewController()), 0)
-	obj, err := storage.Delete(ctx, validController.Name, nil)
-	if err != nil {
-		t.Errorf("unexpected error: %v", err)
-	}
-	if status, ok := obj.(*api.Status); !ok {
-		t.Errorf("Expected status of delete, got %#v", status)
-	} else if status.Status != api.StatusSuccess {
-		t.Errorf("Expected success, got %#v", status.Status)
-	}
-	if len(fakeClient.DeletedKeys) != 1 {
-		t.Errorf("Expected 1 delete, found %#v", fakeClient.DeletedKeys)
-	}
-	if fakeClient.DeletedKeys[0] != key {
-		t.Errorf("Unexpected key: %s, expected %s", fakeClient.DeletedKeys[0], key)
-	}
-}
-
-func TestEtcdWatchController(t *testing.T) {
-	ctx := api.NewDefaultContext()
-	storage, fakeClient := newStorage(t)
-	watching, err := storage.Watch(ctx,
-		labels.Everything(),
-		fields.Everything(),
-		"1",
-	)
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	fakeClient.WaitForWatchCompletion()
-
-	select {
-	case _, ok := <-watching.ResultChan():
-		if !ok {
-			t.Errorf("watching channel should be open")
-		}
-	default:
-	}
-	fakeClient.WatchInjectError <- nil
-	if _, ok := <-watching.ResultChan(); ok {
-		t.Errorf("watching channel should be closed")
-	}
-	watching.Stop()
-}
-
-func TestEtcdWatchControllersMatch(t *testing.T) {
-	ctx := api.WithNamespace(api.NewDefaultContext(), validController.Namespace)
-	storage, fakeClient := newStorage(t)
-	fakeClient.ExpectNotFoundGet(etcdgeneric.NamespaceKeyRootFunc(ctx, "/registry/pods"))
-
-	watching, err := storage.Watch(ctx,
-		labels.SelectorFromSet(validController.Spec.Selector),
-		fields.Everything(),
-		"1",
-	)
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	fakeClient.WaitForWatchCompletion()
-
-	// The watcher above is waiting for these Labels, on receiving them it should
-	// apply the ControllerStatus decorator, which lists pods, causing a query against
-	// the /registry/pods endpoint of the etcd client.
-	controller := &api.ReplicationController{
-		ObjectMeta: api.ObjectMeta{
-			Name:      "foo",
-			Labels:    validController.Spec.Selector,
-			Namespace: "default",
+	test := resttest.New(t, storage, fakeClient.SetError)
+	test.TestWatch(
+		validNewController(),
```

Comment:
Can we use the object validController directly rather than instantiating a new controller?